lib/packwerk/offenses_formatter.rb

```diff
@@ -6,7 +6,39 @@ module OffensesFormatter
     extend T::Sig
     extend T::Helpers
 
-    interface!
+    abstract!
+
+    class << self
+      extend T::Sig
+
+      sig { params(base: Class).void }
+      def included(base)
+        @formatters ||= T.let(@formatters, T.nilable(T::Array[Class]))
+        @formatters ||= []
+        @formatter_by_identifier = nil
```

Comment:
Note that we `nil` out the hash lookup when we add a new formatter. In practice, this should not be necessary, and we can remove if folks want. What's going on here is that a CLI was initialized in test, which included some formatters but not all, and then a later test added new formatters, but they were not properly registered due to the previously memoized lookup hash.

We could just change the tests to only load the CLI after we've loaded configuration, but this felt like an appropriate measure too. Open to changing!